index.js

```diff
@@ -34,7 +34,22 @@ function mkdirp(dir) {
 			});
 	};
 
-	return mkdir(dirs.pop());
+	const checkIfDirExists = dir => {
+		dir = slash(dir);
+
+		return pify(this.raw.mlst.bind(this.raw))(dir)
+			.then(() => dirs.length > 0 && checkIfDirExists(dirs.pop()))
+			.catch(err => {
+				if (err.code !== 550 || dirs.length === 0) {
+					err.message += ` - mkd: ${dir}`;
+					throw err;
+				} else {
```

Comment:
Unnecessary `else` statement since it's implicit that the code will stop executing if throwing.